### PR TITLE
4524 Fix for file update message for netcdf and timeseries aggregation

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a resource
2. Upload netcdf file. A multi-dimensional aggregation will be created. Go to the resource view mode. Verify that there is not a  message at the top of the page - 'One or more NetCDF files are out of sync with metadata changes' Go to the resource edit mode. Click on the netcdf aggregation to view aggregation metadata.  Verify there is not a button to update netcdf file.
3. Edit metadata for netcdf aggregation. Once you save the edited metadata, you should see a button to update the NetCDF file.
4. Upload a SQLite file to create a timeseries aggregation. Go to the resource view mode.  Verify that there is not a message at the top of the page - 'One or more SQLite files are out of sync with metadata changes.' Go to the resource edit mode. Click on the timeseries  aggregation to view aggregation metadata.  Verify that there is not a button to update SQLite file.
5. Edit metadata for timeseries aggregation. Once you save the edited metadata, you should see a button to update the SQLite file.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4543](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4543)

**Comment by pkdash** [comment]
@devincowan It seems you aborted the Jenkins build for this PR.

**Comment by devincowan** [comment]
@pkdash apologies that was an accident!

**Comment by pkdash** [comment]
@devincowan No problem.

File: hs_file_types/models/timeseries.py [hunk 1]
```diff
@@ -1884,6 +1887,9 @@ def sqlite_file_update(instance, sqlite_res_file, user):
         if not instance.has_sqlite_file and instance.can_add_blank_sqlite_file:
             add_blank_sqlite_file(instance, upload_folder='')
         # instance.add_blank_sqlite_file(user)
+    if is_file_type:
+        if not instance.metadata.is_update_file:
+            return
```
Comment (hyi) [inline hs_file_types/models/timeseries.py:1892]: Two input conditions can be combined into one line

File: hs_file_types/views.py [hunk 2]
```diff
@@ -466,9 +466,17 @@ def update_metadata_element(request, hs_file_type, file_type_id, element_name,
...
+
+        if logical_file.type_name() == "NetCDFLogicalFile":
+            logical_file.metadata.is_update_file = True
+            logical_file.metadata.save()
+
         if logical_file.type_name() == "TimeSeriesLogicalFile":
-            ajax_response_data['is_dirty'] = logical_file.metadata.is_dirty
+            logical_file.metadata.is_update_file = True
+            logical_file.metadata.save()
+            ajax_response_data['is_update_file'] = logical_file.metadata.is_update_file
```
Comment (hyi) [inline hs_file_types/views.py:477]: Looks like ajax_response_data needs to be set for NetCDFLogicalFile as well. If true, two conditional checks above can be combined into one by checking type_name is NetCDF or TimeSeries logical file.

File: hs_file_types/views.py [hunk 3]
```diff
@@ -538,9 +546,16 @@ def add_metadata_element(request, hs_file_type, file_type_id, element_name, **kw
...
             ajax_response_data['can_update_sqlite'] = logical_file.can_update_sqlite_file
```
Comment (hyi) [inline hs_file_types/views.py:557]: Same comment as above to combine two conditional checks into one.

**Review by hyi — COMMENTED** [review]
@pkdash Left some comments for your consideration.

File: hs_file_types/views.py (hunk 2 above)
Comment (pkdash) [reply hs_file_types/views.py]: Fixed

File: hs_file_types/views.py (hunk 3 above)
Comment (pkdash) [reply hs_file_types/views.py:553]: Fixed

**Comment by pkdash** [comment]
@hyi See if I have addressed your comments.

**Review by hyi — DISMISSED** [review]

**Comment by pkdash** [comment]
@hyi I had to resolve migration conflicts. Can you re-approve?

**Review by hyi — APPROVED** [review]

**Pull request merged**